### PR TITLE
T23534: Rename project to libanimation

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T23534

### CODE REVIEW

**Comment by smspillaz** [comment]
Argh, looks like I based this work off the wrong fork, hence the mountain of conflicts.

They're resolvable, but I'll need to look into it in the morning, I suspect.

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[submodule "wobbly/third_party/allow_move_optional"]
-	path = wobbly/third_party/allow_move_optional
-	url = git://github.com/smspillaz/CopyMoveConstrainedOptional
```
Comment (smspillaz) [inline .gitmodules]: I should probably split out this change into a separate commit

**Comment by smspillaz** [comment]
Jenkins is failing because it still expects the package name to be `libwobbly` - merging https://github.com/endlessm/eos-build/pull/844 will fix that.

File: animation/geometry.h [hunk 2]
```diff
@@ -337,20 +337,30 @@ namespace animation
...
         bool Box <T>::contains (U const &p) const
         {
-            auto const x1 = dimension::get <0> (tr);
-            auto const x2 = dimension::get <0> (bl);
-            auto const y1 = dimension::get <1> (tr);
-            auto const y2 = dimension::get <1> (bl);
+            auto const x1 = dimension::get <0> (tl);
+            auto const x2 = dimension::get <0> (br);
+            auto const y1 = dimension::get <1> (tl);
+            auto const y2 = dimension::get <1> (br);
```
Comment (joaquimrocha) [inline animation/geometry.h]: If you're changing the corners used for extracting the limits, and you're not changing the comparison, then it means the comparison is wrong now or before. Seems like it was wrong before. Or was it just the name of the vars that was incorrect? If so, then maybe making more emphasis on that fact in the commit message would be good?

File: animation/wobbly/wobbly.cpp [hunk 3]
```diff
@@ -1253,10 +1252,10 @@ wobbly::BezierMesh::Extremes () const
...
-wobbly::PointView <double>
+animation::PointView <double>
 wobbly::BezierMesh::PointForIndex (size_t x, size_t y)
 {
-    return wobbly::PointView <double> (mPoints,
-                                       CoordIndex (x, y,
-                                                   config::Width));
+    return animation::PointView <double> (mPoints,
+                                          CoordIndex (x, y,
+                                                      config::Width));
```
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation/wobbly/wobbly.cpp]: It isn't. Well, it used to be but I ditched that rule since it was getting too unwieldy to comply with :)

File: animation-glib/wobbly/anchor.cpp [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+static void
+animation_wobbly_anchor_class_init (AnimationWobblyAnchorClass *klass)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  object_class->finalize = animation_wobbly_anchor_finalize;
+}
+
+AnimationWobblyAnchor *
+animation_wobbly_anchor_new_for_native_anchor_rvalue (wobbly::Anchor &&anchor)
```
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation-glib/wobbly/anchor.cpp]: Probably not, though I guess I just wanted to make it clear that it wants an rvalue reference in case anyone was wondering why things might not compile when they pass it an lvalue :)

File: animation-glib/meson.build [hunk 5]
```diff
@@ -15,68 +17,60 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
-#
-# Build the libwobbly library.
 
-api_version = '0.3'
+api_version = '0'
```
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation-glib/meson.build]: Yeah the API version got reset to 0 since we changed the name. Or at least I assume that's how things are meant to work.

File: animation/geometry.h (hunk 2 above)
Comment (smspillaz) [reply animation/geometry.h]: In this case, the values themselves didn't actually change, only the names, which were wrong. I'll re-emphasize that in the commit message.

File: animation/wobbly/wobbly_internal.h [hunk 6]
```diff
@@ -44,9 +44,9 @@
 #include <math.h>                       // for fabs
 #include <stddef.h>                     // for size_t
 
-#include <animation/wobbly/geometry.h>                   // for PointView, PointModel, etc
-#include <animation/wobbly/geometry_traits.h>            // for assign, scale, etc
-#include <animation/wobbly/wobbly.h>    // for PointView, Vector, Point, etc/wobbly_internal.h
+#include <animation/geometry.h>                   // for PointView, PointModel, etc
+#include <animation/geometry_traits.h>            // for assign, scale, etc
+#include <animation/wobbly/wobbly.h>    // for PointView, Vector, Point, etc
```
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation/wobbly/wobbly_internal.h]: :P

[the remark this answers is not in the document]
Comment (smspillaz) [reply animation/wobbly/wobbly_internal.h]: Fixed

File: animation/geometry.h (hunk 2 above)
Comment (smspillaz) [reply animation/geometry.h]: Commit message fixed.

File: animation/wobbly/wobbly.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation/wobbly/wobbly.cpp]: Fixed.

File: animation/wobbly/wobbly.cpp [hunk 7]
```diff
@@ -605,29 +604,29 @@ namespace
...
+        agd::assign (start, anchor);
 
         size_t const row = index % wobbly::config::Width;
         size_t const column = index / wobbly::config::Width;
 
-        wobbly::Point deltaToTopLeft (wgd::get <0> (tileSize) * row,
-                                      wgd::get <1> (tileSize) * column);
-        wgd::pointwise_subtract (start, deltaToTopLeft);
+        animation::Point deltaToTopLeft (agd::get <0> (tileSize) * row,
+                                      agd::get <1> (tileSize) * column);
```
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation/wobbly/wobbly.cpp]: Fixed

File: animation-glib/wobbly/anchor.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (smspillaz) [reply animation-glib/wobbly/anchor.cpp]: I left this as is for now
